### ISSUE TITLE
Block Scripts under  Brave Shields is not working

### ISSUE THREAD

**Issue by soner-yuksel** [brave/brave-ios#6724]
### Description: 

**Block Scripts** shields is not working. Both changing the shields from Shields screen or under shields defaults in settings doesnt work and javascript is not blocked on website.

There are also reports of this behaviour from users https://www.reddit.com/r/brave_browser/comments/106xpxi/block_scripts_not_working_anymore_ios/

### Steps to Reproduce 
  1. Load a website that can trigger a javascript action or inject a script
  2. Test the script is performing the desired action
  3. Change the Block Scripts value on/off
  4. Test and validate if script is working or blocked

### Actual result: 

Javascript blocking is not working

### Expected result:

When Block Scripts is enabled the javascript actions should be blocked

### Reproduces how often: [Easily reproduced, Intermittent Issue]

Easily reproduced

### Brave Version: 

- Can you reproduce this issue with the most recent build from TestFlight? Yes
- Can you reproduce this issue with the previous version of the current build from TestFlight? Yes
- Can you reproduce this issue with the current build from AppStore? Yes

### Device details: 
N/A

### Website problems only:
- did you check with Brave Shields down? Yes
- did you check in Safari/Firefox (WkWebView-based browsers)? Yes

**Comment by diracdeltas** [comment]
this is a major regression. can we hotfix + add QA or automated tests to make sure it doesn't regress further?

**Comment by iccub** [comment]
cc @kjozwiak

**Comment by kjozwiak** [comment]
As per the Slack discussion, we need to get this out via a HF ASAP. It will be a while till we get `1.47.x` out the door so we can't wait that long. Moving this into the https://github.com/brave/brave-ios/milestone/105 for now. @iccub @Brandon-T @soner-yuksel could we get this merged into `1.46.x` and get a TF build going?

**Comment by kjozwiak** [comment]
@Uni-verse  a good case that we can use for the above is what we use on `Desktop`:

* `Visit https://www.blizzard.com/ and then turn on script blocking, page should not load.`

**Comment by soner-yuksel** [comment]
Or if you guys want direct visual response for now you can use 

`https://www.whatismybrowser.com/detect/is-javascript-enabled`

**Comment by kjozwiak** [comment]
Reproduced the original issue on `iPhone 11` running `iOS 16.2` using `1.46.1 (23.1.5.18)` as per the following:

* visited https://www.blizzard.com and enabled `Block Scripts` via the shields panel
* notice that the content in the carousel isn't being blocked

https://user-images.githubusercontent.com/2602313/211661327-6deafb2a-d362-4b52-9433-a71343eaf3f6.mp4

Verification **PASSED** on `iPhone 11` running `iOS 16.2` using `1.46.2 (23.1.10.20)`

**`Upgrade Case/Block Scripts Enabled`**

* downloaded/installed `1.46.1 (23.1.5.18)`
* visited https://www.blizzard.com and enabled `Block Scripts` via the shields panel (notice it's not working as expected)
* updated to `1.46.2 (23.1.10.20)` via TF and visited https://www.blizzard.com (ensured that the carousel content is blocked)

**`Upgrade Case/Block Scripts Disabled`**

* downloaded/installed `1.46.1 (23.1.5.18)`
* updated to `1.46.2 (23.1.10.20)` via TF
* visited https://www.blizzard.com and enabled `Block Scripts` via the shields panel
* ensured that the content via the carousel being blocked
* ensured that enabling/disabling `Block Scripts` is working as expected (content is being blocked)

**`Clean Install/Profile`**

* installed `1.46.2 (23.1.10.20)` via TF
* visited https://www.blizzard.com and enabled `Block Scripts` via the shields panel
* ensured that the content within the carousel is being blocked
* ensured that enabling/disabling `Block Scripts` is working as expected (content is being blocked)
* visited https://www.whatismybrowser.com/detect/is-javascript-enabled and ensured the correct state is being listed

**`Example (JS Enabled)`** | **`Example (JS Enabled)`**
------------------------------|------------------------------
![IMG_3903](https://user-images.githubusercontent.com/2602313/211662123-448dfdaf-1b88-42e8-aa2a-2e000dc04edd.PNG) | ![IMG_3904](https://user-images.githubusercontent.com/2602313/211662132-4011ef62-70b4-4573-8205-95a69a6ce16b.PNG)

**`Example (JS Disabled)`** | **`Example (JS Disabled)`**
------------------------------|------------------------------
![IMG_3905](https://user-images.githubusercontent.com/2602313/211662211-a5ba8b55-f40a-450f-a2b0-b31a8aa32ff6.PNG) | ![IMG_3906](https://user-images.githubusercontent.com/2602313/211662239-f59954a7-bf29-4166-bb9d-4284e3078f5f.PNG)

https://user-images.githubusercontent.com/2602313/211661664-f59031a2-1049-4bb6-b815-8fdce769624c.mp4

**Comment by Uni-verse** [comment]
Verified on `iPad (9th Gen)` running `iPad OS 15.6.1` on build version `1.46.2 (23.1.10.20)`

**Fresh Profile**

- Verified enabling `Block Scripts` in shields will disable javascript when on webpages (`blizzard.com`)
- Verified disabling `Block Scripts` from shields will enable javascript 
- Verified enabling `Block Scripts` in global privacy settings will be applied on websites and can be disabled per site from shields

Block Scripts Disabled | Block Scripts Enabled
-|-
![IMG_0089](https://user-images.githubusercontent.com/17885425/211669435-f3df592d-eb09-4c65-a000-ebd868bd68c4.PNG) | ![IMG_0088 2](https://user-images.githubusercontent.com/17885425/211669425-f59b5a45-0af3-4666-b7ba-c713b79549f5.PNG)
![IMG_0092 2](https://user-images.githubusercontent.com/17885425/211669443-0d7aeca4-6b6f-4e15-b203-cb32870d74f6.PNG) | ![IMG_0091](https://user-images.githubusercontent.com/17885425/211669439-d7cfd95f-60e6-4a0c-9365-4c3a2e8446f9.PNG)


https://user-images.githubusercontent.com/17885425/211670413-ea42a615-0fa8-4f06-a85f-b35f5921c212.MP4

**Upgraded Profile - Settings retained (1.46.1 -> 1.46.2)** 

1. Install 1.46.1
2. Load twitter.com
3. Enabled `Block Scripts` from Shields settings
4. Upgrade to 1.46.2
5. Check Block Scripts setting in Shields to ensure it is pretained from previous version and javascript is disabled on the page.

1.46.1 | 1.46.2
-|-
![IMG_0095 3](https://user-images.githubusercontent.com/17885425/211673543-1c890df9-1dfb-4e3e-96f5-bec3d64ec597.PNG) | ![IMG_0097 2](https://user-images.githubusercontent.com/17885425/211673552-7ca1cacf-f4d0-46fe-96c5-0861e9c845cf.PNG)



**Upgraded Profile (1.46.1 -> 1.46.2)**

- Verified enabling `Block Scripts` in shields will disable javascript when on webpages (`blizzard.com`)
- Verified disabling `Block Scripts` from shields will enable javascript 
- Verified enabling `Block Scripts` in global privacy settings will be applied on websites and can be disabled per site from shields

Block Scripts Enabled | Block Scripts Disabled | Block Scripts Enabled
-|-|-
![IMG_0088 2](https://user-images.githubusercontent.com/17885425/211671207-aa172291-e0dd-4d87-99c0-8aa2c43bec31.PNG) | ![IMG_0089](https://user-images.githubusercontent.com/17885425/211671209-cb315dd0-b1a1-4892-b52f-6252fb36265e.PNG) | ![IMG_0091](https://user-images.githubusercontent.com/17885425/211671212-ee46f4b8-8035-404a-9b5e-7ab713622144.PNG)

**Comment by Uni-verse** [comment]
Verified on `iPhone X` running `iOS 14.4.2` on build version `1.46.2 (23.1.10.20)`

- Verified `Block Scripts` shield setting is retained per site on `upgraded profile`
- Verified Javascript is disabled when enabled `Block Scripts` from shields at `Blizzard.com` on fresh profile as well as upgraded profile
- Verified enabling global Block Script settings works to disable JS on websites and can be disabled per site from shields menu

1.46.1 | 1.46.2
-|-
![IMG_2829](https://user-images.githubusercontent.com/17885425/211675207-5ea76d3a-c867-4399-951b-ca579859dfdf.PNG) | ![IMG_2832](https://user-images.githubusercontent.com/17885425/211675430-2d6eed1b-237b-4054-b7ae-739fa4ea35d8.PNG)
![IMG_2830](https://user-images.githubusercontent.com/17885425/211675211-3a860364-b081-46cb-8941-c61b4ee3d857.PNG) | ![IMG_2831](https://user-images.githubusercontent.com/17885425/211675424-e8893b25-871d-440c-a8b6-f380d40372be.PNG)
